common/src/main/java/org/apache/cassandra/sidecar/common/NodeSettings.java

```diff
@@ -57,23 +82,41 @@ public String partitioner()
         return partitioner;
     }
 
+    @JsonProperty("sidecar")
+    public Map<String, String> sidecar()
+    {
+        return sidecar;
+    }
+
+    public String sidecarVersion()
+    {
+        return sidecar.get(VERSION);
```

Comment:
can you do a null check for `sidecar`? It is prone to NPE.

Reply:
Done.